tests/test_freezable.py

```diff
@@ -0,0 +1,72 @@
+"""Testing for ElfPy types"""
+from __future__ import annotations  # types are strings by default in 3.11
+
+import unittest
+from dataclasses import dataclass
+
+from elfpy.types import freezable
+
+
+@freezable
+@dataclass
+class FreezableClass:
+    """Freezable class for testing"""
+
+    def __init__(self):
```

Comment:
No need to define init on a dataclass. In fact I'm surprised it let you at all. Just set it as a member attribute:

```
@freezable
@dataclass
class FreezableClass:
    """Freezable class for testing"""
    
    existing_attrib: int = 1
```